prusti-viper/src/encoder/encoder.rs

```diff
@@ -382,11 +382,16 @@ impl<'v, 'r, 'a, 'tcx> Encoder<'v, 'r, 'a, 'tcx> {
             // TODO(@jakob): fix reliance on self
             if let Some(id) = self.env().tcx().trait_of_item(proc_def_id) {
                 let proc_name = self.env().tcx().item_name(proc_def_id).as_symbol();
-                if let Some(item) = self.env().get_trait_method_decl_for_type(ty, id, proc_name) {
-                    if let Some(spec) = self.get_spec_by_def_id(item.def_id) {
-                        impl_spec = spec.clone();
-                    } else {
-                        debug!("Procedure {:?} has no specification", item.def_id);
+                let proc_res = self.env().get_trait_method_decl_for_type(ty, id, proc_name);
+                if proc_res.is_err() {
+                    info!("trait refinement not supported on generic traits");
```

Comment:
It would be great if you add a variant to `EncodingError` (even just a `Generic(String)` one) and call `self.register_encoding_error(..)` here, instead of `info!(..)`.

Reply:
Sure, but this will make some existing tests fail. Some tests implement traits with different associated items, and can be verified, as they don't rely on contract refinement on those trait methods. If I register an error there, it will fail compilation and therefore fail the test.

See test `008-num-traits-pow.rs`.

Reply:
I will add the error, as I guess it would be useful also for other parts of the code, and push it here. Then we can see which tests will fail.